src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -1126,7 +1126,7 @@ private BoundLambda ReallyBindForErrorRecovery()
                 ?? rebind(ReallyInferReturnType(delegateType: null, ImmutableArray<TypeWithAnnotations>.Empty, ImmutableArray<RefKind>.Empty));
 
             // Rebind a lambda to push target conversions through the return/result expressions
-            [return: NotNullIfNotNull("lambda")] BoundLambda? rebind(BoundLambda? lambda)
+            [return: NotNullIfNotNull(nameof(lambda))] BoundLambda? rebind(BoundLambda? lambda)
```

Comment:
❓ What's the scope here if a local variable has the same name?

Reply:
it's only the parameters.